lib/posix/posix_internal.h

```diff
@@ -105,4 +105,15 @@ struct posix_thread *to_posix_thread(pthread_t pth);
 /* get and possibly initialize a posix_mutex */
 struct k_mutex *to_posix_mutex(pthread_mutex_t *mu);
 
+/* TODO: PID to TID conversion assumes two design decisions:
+ *  1) each process has its main thread with pthread_t TID;
+ *  2) PID value is assigned as the main thread's TID which
+ *     must be already initialized whereas valid PID should
+ *     have positive value.
+ */
+static inline pthread_t z_get_tid(pid_t pid)
```

Comment:
Personally, I would welcome proper support for separate Process IDs and processes done the right way at the kernel level, but it's probably not a good idea to try and implement a PID structure outside of the kernel.

Currently, we can only assume there is at most 1 process ID (maybe 2 in a sense) and that Zephyr runs in a single-process environment.

If you have plans to add that functionality to the kernel, it would be quite important to submit that as an RFC first.
https://github.com/zephyrproject-rtos/zephyr/issues/new/choose

Reply:
> Personally, I would welcome proper support for separate Process IDs and processes done the right way at the kernel level, but it's probably not a good idea to try and implement a PID structure outside of the kernel.

I agree, that's why I'm trying to have a single wrapper function here at POSIX API level which can be later adjusted to whatever decision (or not ?) will be at the Zephyr kernel in regard of Process object and its PID going in the opposite way from POSIX standard's history perspective.

There is a bunch of POSIX APIs with `pid_t` parameters and in case of `sched_*` they need to be related to a thread, so we can assume for now that PID value maps 1:1 to the 'main' thread's id.

Reply:
> > will be at the Zephyr kernel in regard of Process object and its PID going in the opposite way from POSIX standard's history perspective.
> 
> Can you please clarify?

well, I just mean that threads were accepted into POSIX after processes, as an extension, and now for Zephyr it is a kind of opposite task to implement processes having threads already.


> I get the impression that your reference in this PR is the Linux implementation of these calls.
> 
> https://linux.die.net/man/2/sched_getscheduler
> 
> Is that correct? If so, then this PR makes significantly more sense

[GNU libc](https://www.gnu.org/software/libc/manual/html_node/Process-Identification.html), Linux NTPL, and the [APUE book](https://en.wikipedia.org/wiki/Advanced_Programming_in_the_Unix_Environment) 

really, I had no intention to suggest any solution for Zephyr process design in this tiny PR )), rather a proposal to discuss it and possibly do something meaningful with these sched_get* functions instead of simply return 'not-yet-implemented'.